src/catkin_pkg/package.py

```diff
@@ -804,6 +809,23 @@ def _get_node_value(node, allow_xml=False, apply_str=True):
     return value
 
 
+def _get_node_text(node, strip=True):
+    value = ''
+    for child in node.childNodes:
+        if child.nodeType == child.TEXT_NODE:
+            value += re.sub('\s+', ' ', child.data)
+        elif child.nodeType == child.ELEMENT_NODE:
+            if child.tagName == 'br':
+                value += '\n'
+            else:
+                value += _get_node_text(child, strip=False)
+        else:
+            assert 'unreachable'
+    if strip:
+        value = value.strip()
+    return value
```

Comment:
Maybe something like this could help? As I understand it, the only way for a newline to show up is when `<br/>` is used, so this should hold for all cases.
```suggestion
    return re.sub('\s?\n\s?', '\n', value)
```

Reply:
This is actually really close to what I ended up with in https://github.com/ros-infrastructure/catkin_pkg/pull/329/commits/b251bb7ee1914d16a443b0d93639dfe51b307872. I implemented it in _only_ for the plaintext_description as I don't think it's generally "less correct", I also made sure to preserve the number of consecutive newlines.